File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n == 1) return 1; 
        if (n == 2) return 2; 
        int prev1 = 2, prev2 = 1, current;
        for (int i = 3; i <= n; i++) {
            current = prev1 + prev2;
            prev2 = prev1;  
            prev1 = current; 
        }
    return prev1;
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def __init__(self):
        self.dp = {}

    def climbStairs(self, n):
        if n <= 2:
            return n
        if n in self.dp:
            return self.dp[n]  # Cached result
        
        # Recursively calculate the number of ways
        self.dp[n] = self.climbStairs(n - 1) + self.climbStairs(n - 2)
        return self.dp[n]
